### PR TITLE
add regularizer api

### PR DESCRIPTION
**Pull request by littletomatodonkey**
### PR types
Function optimization

### PR changes
APIs

### Describe
1. Add `paddle.regularizer.L1Decay`, `paddle.regularizer.L2Decay` API.
2. Add `paddle.utils.unique_name`, `paddle.utils.load_op_library` and `paddle.utils.require_version`.

### CODE REVIEW

**Review by wawltor — DISMISSED** [review]
LGTM

File: python/paddle/__init__.py [hunk 1]
```diff
@@ -49,6 +49,7 @@
 import paddle.metric
 import paddle.device
 import paddle.incubate.complex as complex
+import paddle.regularizer
```
Comment (kinghuin) [inline python/paddle/__init__.py:52]: 其中的regularization_coeff需要优化为coeff

Comment (littletomatodonkey) [reply python/paddle/__init__.py:52]: done, thx

File: python/paddle/regularizer.py [hunk 2]
```diff
@@ -12,8 +12,152 @@
...
+                    bias_attr=False)
+    """
+
+    def __init__(self, coeff=0.0):
+        super(L1Decay, self).__init__(coeff)
+
+
+class L2Decay(fluid.regularizer.L2Decay):
+    """
+    Implement the L2 Weight Decay Regularization, which encourages the weights to be sparse.
```
Comment (jzhang533) [inline python/paddle/regularizer.py]: 这里描述不对，跟L1 decay的一样了。
L2并不会弄成sparse weights

File: python/paddle/regularizer.py [hunk 3]
```diff
@@ -12,8 +12,152 @@
...
+                                             parameters=net.parameters(),
+                                             weight_decay=L1Decay(0.001))
+                for epoch in range(epochs):
+                    for batch_id, data in enumerate(train_loader()):
+                        x_data = data[0]
+                        y_data = data[1]
+                        predicts = net(x_data)
+                        loss = paddle.nn.functional.cross_entropy(predicts, y_data)
+                        # calc loss
+                        acc = paddle.metric.accuracy(predicts, y_data, k=2)
```
Comment (jzhang533) [inline python/paddle/regularizer.py]: 注释是calc loss
紧接着一行却是计算accuracy
这个示例代码应该可以再简洁一些。

File: python/paddle/regularizer.py [hunk 4]
```diff
@@ -12,8 +12,152 @@
...
+
+        L2WeightDecay = reg\_coeff * parameter
+
+    Args:
+        regularization_coeff(float, optional): regularization coeff. Default:0.0
+	
+    Examples:
+        .. code-block:: python
+
+            # Example1: set Regularizer in optimizer
```
Comment (jzhang533) [inline python/paddle/regularizer.py:107]: 示例代码可以更简洁一些。

Comment (jzhang533) [inline python/paddle/regularizer.py:107]: 示例代码可以更简洁一些。

File: python/paddle/regularizer.py [hunk 5]
```diff
@@ -12,8 +12,152 @@
...
+            from paddle import ParamAttr
+            from paddle.regularizer import L2Decay
+
+            my_conv2d = Conv2d(
+                    in_channels=10,
+                    out_channels=10,
+                    kernel_size=1,
+                    stride=1,
+                    padding=0,
+                    weight_attr=ParamAttr(regularizer=L2Decay(coeff=0.0)),
```
Comment (jzhang533) [inline python/paddle/regularizer.py]: this is L1
and use a non zero coeff setting

**Review by jzhang533 — COMMENTED** [review]
示例代码建议简洁一些。

File: python/paddle/regularizer.py (hunk 4 above)
Comment (littletomatodonkey) [reply python/paddle/regularizer.py:107]: 已修改~

File: python/paddle/regularizer.py (hunk 5 above)
Comment (littletomatodonkey) [reply python/paddle/regularizer.py]: 已修改~

File: python/paddle/regularizer.py (hunk 4 above)
Comment (littletomatodonkey) [reply python/paddle/regularizer.py:107]: 已修改~

File: python/paddle/regularizer.py (hunk 3 above)
Comment (littletomatodonkey) [reply python/paddle/regularizer.py]: 已修改~

File: python/paddle/regularizer.py (hunk 2 above)
Comment (littletomatodonkey) [reply python/paddle/regularizer.py]: 已修改~

File: python/paddle/regularizer.py [hunk 6]
```diff
@@ -12,8 +12,130 @@
...
+
+
+class L1Decay(fluid.regularizer.L1Decay):
+    """
+    Implement the L1 Weight Decay Regularization, which encourages the weights to be sparse.
+    
+    It can be set in :ref:`api_fluid_ParamAttr` or ``optimizer`` (such as :ref:`api_paddle_optimizer_Momentum` ). 
+    When set in ``ParamAttr`` , it only takes effect for trainable parameters in this layer. When set in 
+    ``optimizer`` , it takes effect for all trainable parameters. When set together, ``ParamAttr`` has 
+    higher priority than ``optimizer`` .
```
Comment (jzhang533) [inline python/paddle/regularizer.py]: higher priority 的意思是说optimizer里的设置就不起作用了吗？

Comment (littletomatodonkey) [reply python/paddle/regularizer.py]: 是的， 可以参考这里的FAQ：https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/faq/train_cn.html#optimizerparamattr
已经添加英文注释了~感谢提醒

**Review by jzhang533 — DISMISSED** [review]
lgtm

**Review by tink2123 — COMMENTED** [review]
LGTM

**Review by lanxianghit — APPROVED** [review]

**Review by jzhang533 — APPROVED** [review]
lgtm

**Review by shippingwang — APPROVED** [review]
LGTM!
